app/api/jobs/routine/[id]/executions/route.ts

```diff
@@ -82,11 +78,25 @@ export async function GET(
             : "unknown",
       })) || [];
 
-    const totalCount = totalExecutions.results?.length || 0;
+    const uniqueEnvs = Array.from(
+      new Set(allTransformedExecutions.map((e) => e.environment))
+    ).sort();
+
+    const filteredExecutions = envFilter
+      ? allTransformedExecutions.filter((e) => e.environment === envFilter)
+      : allTransformedExecutions;
+
+    const paginatedExecutions = filteredExecutions.slice(
+      offset,
+      offset + pageSize
+    );
+
+    const totalCount = filteredExecutions.length;
 
     return NextResponse.json({
-      executions: transformedExecutions,
+      executions: paginatedExecutions,
```

Comment:
For some reason, I'm getting no executions returned, please revise

<img width="632" height="284" alt="Image" src="https://github.com/user-attachments/assets/20ffb95a-eaa4-4f23-9781-c6ff3166e682" />